### PR TITLE
PimSetExtends operation implemented

### PR DESCRIPTION
**Pull request by sstenchlak**

### CODE REVIEW

File: lib/core/store/federated-store/read-only-federated-store.ts [hunk 1]
```diff
@@ -40,7 +40,7 @@ export class ReadOnlyFederatedStore implements CoreResourceReader {
   async listResourcesOfType(typeIri: string): Promise<string[]> {
     const resources = new Set<string>();
     for (const model of this.readers) {
-      (await model.listResourcesOfType(typeIri)).forEach(resources.add);
+      (await model.listResourcesOfType(typeIri)).forEach(r => resources.add(r));
```
Comment (skodapetr) [inline lib/core/store/federated-store/read-only-federated-store.ts:43]: Why is this change needed?

Comment (sstenchlak) [reply lib/core/store/federated-store/read-only-federated-store.ts:43]: First of all, this has nothing to do with the second commit here. I was just lazy to create two separate PR. Sorry. The problem is that the internal implementation of `Set` is probably done by function, not by arrow function. Therefore it has a dynamic execution context, and in my use case, it didn't work as expected.

Comment (skodapetr) [reply lib/core/store/federated-store/read-only-federated-store.ts:43]: Could you specify the use-case? It would be best to create a test that would fail with the original implementation but pass with the updated one.

Comment (sstenchlak) [reply lib/core/store/federated-store/read-only-federated-store.ts:43]: Well, I think this is not worth testing (of course, I can create a test if you want) because apparently, this is the "how you should not program" kind of bug rather something that can be a problem specifically in this scenario.

Minimal example would be
```js
const set = new Set();
[1, 2, 3].forEach(n => set.add(n));
[4, 5, 6].forEach(set.add); // TypeError: Method Set.prototype.add called on incompatible receiver undefined
```
